setup/bundles.rst

```diff
@@ -198,6 +198,56 @@ adding a ``setDefined()`` method. The recommended check in this case would be::
     to detect the version of the ``symfony/http-kernel`` component, because it
     is the component where this constant is defined.
 
+Generic conditions for component version detection
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You should strive to detect the exact feature that you want to use as
+soon as it is available. That is not always possible however, so here is
+a list of generic conditions you may use for lack of a better
+alternative. These conditions detect new APIs as soon as they are
+available.
+
+2.6
+...
+
+OptionsResolver
+"""""""""""""""
+
+.. code-block:: php
+
+    method_exists(Symfony\Component\OptionsResolver\OptionsResolver, 'setDefined')
+
+2.7
+...
+
+Form
+""""
+
+.. code-block:: php
+
+    method_exists('Symfony\Component\Form\AbstractType', 'configureOptions')
+
+2.8
+...
+
+Form
+""""
+
+.. code-block:: php
+
+    method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')
+
+3.0
+...
+
+Form
```

Comment:
If this should be documented, this should be done for every component as well, and all their API, not sure this worth it. It should be done on a case by case basis, so leading the way as this was done before should be enough.